### DISCUSSION TITLE
Mattermost webhook

### DISCUSSION THREAD

**Discussion by JeanYvesQueignec** [fluxcd/flux2#1645] [Q&A]
Hi,
First of all , thanks for this nice tool !  I  have just started to test it and It looks impressive.
I get a point with a  question related to Mattermost : Is it possible to add it as a notification provider ?
Thanks,
Regards,
Jean-Yves

**Comment by yebyen** [comment]
According to this issue, Mattermost is already supported via the Slack integration: https://github.com/fluxcd/notification-controller/issues/203

This thread in our Slack channel indicated there are some quirks: https://cloud-native.slack.com/archives/CLAJ40HV3/p1626348086356900?thread_ts=1626348086.356900&cid=CLAJ40HV3

The full text of that comment, (in case this link doesn't work, I typically don't have great luck with Slack links):

> Notification Controller Findings:

> I have been trying to implement the notification controller and I have been getting the following error:
> {"level":"info","ts":"2021-07-13T17:27:35.792Z","logger":"event-server","msg":"Dispatching event: no updates made","reconciler kind":"ImageUpdateAutomation","name":"microservices","namespace":"flux-system"}
> {"level":"error","ts":"2021-07-13T17:27:35.873Z","logger":"event-server","msg":"failed to send notification","reconciler kind":"ImageUpdateAutomation","name":"microservices","namespace":"flux-system","error":"postMessage failed: request failed with status code 404, {\"id\":\"web.incoming_webhook.channel.app_error\",\"message\":\"Couldn't find the channel.\",\"detailed_error\":\"\",\"request_id\":\"nzu7ixx7itb8pycxysghkc3reo\",\"status_code\":404}"}
> I was trying to implement the notification controller to work with Mattermost webhooks.
> After a couple days of troubleshooting and changing things around I figured out that my issues was with how I created my webhook. I created the webhook to strictly work in the channel I specified. By removing the "channel" tag in the provider yaml file, I was able to get the notification controller to work as specified in the documentation.
> My theory is that the process was trying to do the following: Mattermost -> "Channel" -> "Channel"

**Reply by yebyen** [reply]
I don't have access to a Mattermost instance to test against, but if you are able to make this work, and willing to take a shot at documenting this in a PR against fluxcd/website, I'm sure that would be of value!
